File: src/Extensibility/SharpDetect.Plugins/Descriptors/MethodInjectionDescriptor.cs
// Copyright 2026 Andrej Čižmárik and Contributors
// SPDX-License-Identifier: Apache-2.0

using SharpDetect.Core.Events;

namespace SharpDetect.Plugins.Descriptors;

public sealed record MethodInjectionDescriptor(
    string Name,
    RecordedEventType EventType,
    MethodSignatureDescriptor Signature);

File: src/SharpDetect.Profiler/LibDescriptors/MethodInjectionDescriptor.cpp
// Copyright 2026 Andrej Čižmárik and Contributors
// SPDX-License-Identifier: Apache-2.0

#include "MethodInjectionDescriptor.h"

void Profiler::to_json(nlohmann::json& json, const MethodInjectionDescriptor& descriptor)
{
	json["name"] = descriptor.name;
	json["eventType"] = descriptor.eventType;
	json["signature"] = descriptor.signature;
}

void Profiler::from_json(const nlohmann::json& json, MethodInjectionDescriptor& descriptor)
{
	descriptor.name = json.at("name");
	descriptor.eventType = json.at("eventType");
	descriptor.signature = json.at("signature");
}